fdbclient/ServerKnobs.cpp

```diff
@@ -651,7 +651,7 @@ void ServerKnobs::initialize(Randomize randomize, ClientKnobs* clientKnobs, IsSi
 	init( CC_WORKER_HEALTH_CHECKING_INTERVAL,                   60.0 );
 	init( CC_DEGRADED_LINK_EXPIRATION_INTERVAL,                300.0 );
 	init( CC_MIN_DEGRADATION_INTERVAL,                         120.0 );
-	init( ENCRYPT_KEY_PROXY_FAILURE_TIME,                        0.1 );
+	init( ENCRYPT_KEY_PROXY_FAILURE_TIME,                        0.1 ); if ( isSimulated ) ENCRYPT_KEY_PROXY_FAILURE_TIME = 1.0 + deterministicRandom()->random01();
```

Comment:
Why don't we just do 1.0 like all of the other singletons and the worker failure time is? Why was it originally 0.1?

Reply:
EKP availability seems to play significant impact to overall cluster availability, hence, 100ms failure detection is better than 1 second failure detection for production environment.
However, in simulation having a much longer failure time detection seems fine.

Reply:
I mean I agree in theory, but if all of our other components which are also critical to cluster availability use 1s instead of 100ms then it's not like this would often be the bottleneck in detecting and initiating a recovery anyway. Plus, our average recovery time is at least several seconds iirc so 100ms vs 1 second doesn't necessarily make or break that either.
My worry in particular is that all other components might use 1s instead of something lower like 100ms because of the occasional occurrence of false positives, which in particular for the EKP would be bad because switching EKPs while the clsuter is healthy has a large performance penalty due to flushing the cache.
Maybe someone like @sfc-gh-etschannen or @sfc-gh-ajbeamon has more context here?

Reply:
I see some knobs which are much more sensitive that 1 second, for instance `CC_THROTTLE_SINGLETON_RERECRUIT_INTERVAL = 500ms`, `MASTER_FAILURE_SLOPE_DURING_RECOVERY = 100ms` and `MASTER_FAILURE_REACTION_TIME = 400ms`

I'm not aware of reasoning of selecting 1 second as a good failure-detection for all singletons, however, given Master failure reaction time is 400 ms, not sure 1 second for all sounds reasonable. Also, not all singletons serve that critical purpose, for instance, RateKeeper taking few unavailability might be OK in most of the cases, same for DD, BlobMaster etc. 

Again, eager to learn how these numbers are chosen, unfortunately we don't have much code documentation in case they are empirically determined running in production environment.